core/src/main/java/com/klaytn/caver/abi/ABI.java

```diff
@@ -0,0 +1,125 @@
+package com.klaytn.caver.abi;
+
+import com.klaytn.caver.contract.ContractEvent;
+import com.klaytn.caver.contract.ContractIOType;
+import com.klaytn.caver.contract.ContractMethod;
+import org.web3j.abi.FunctionEncoder;
+import org.web3j.abi.FunctionReturnDecoder;
+import org.web3j.abi.TypeEncoder;
+import org.web3j.abi.TypeReference;
+import org.web3j.abi.datatypes.*;
+import org.web3j.crypto.Hash;
+import org.web3j.utils.Numeric;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ABI {
+    public static String encodeFunctionCall(ContractMethod method, List<Type> params) {
+        String methodId = encodeFunctionSignature(method);
+        String encodedParams = encodeParameter(params);
+
+        return methodId + encodedParams;
+    }
+
+    public static String encodeFunctionCall(String functionSig, List<Type> params) {
+        String methodId = encodeFunctionSignature(functionSig);
+        String encodedParams = encodeParameter(params);
+
+        return methodId + encodedParams;
+    }
+
+    public static String encodeFunctionSignature(ContractMethod method) {
+        return encodeFunctionSignature(buildFunctionSignature(method));
+    }
+
+    public static String encodeFunctionSignature(String functionSig) {
+        byte[] input = functionSig.getBytes();
+        byte[] hash = Hash.sha3(input);
+        return Numeric.toHexString(hash).substring(0, 10);
+    }
+
+    public static String buildFunctionSignature(ContractMethod method) {
+        StringBuilder result = new StringBuilder();
+        result.append(method.getName());
+        result.append("(");
+        String params = method.getInputs().stream()
+                .map(ContractIOType::getType)
+                .collect(Collectors.joining(","));
+        result.append(params);
+        result.append(")");
+
+        return result.toString();
+    }
+
+    public static String encodeEventSignature(ContractEvent event) {
+        return encodeEventSignature(buildEventSignature(event));
+    }
+
+    public static String encodeEventSignature(String eventSig) {
+        byte[] input = eventSig.getBytes();
+        byte[] hash = Hash.sha3(input);
+        return Numeric.toHexString(hash);
+    }
+
+    public static String buildEventSignature(ContractEvent method) {
+        StringBuilder result = new StringBuilder();
+        result.append(method.getName());
+        result.append("(");
+        String params = method.getInputs().stream()
+                .map(ContractIOType::getType)
+                .collect(Collectors.joining(","));
+        result.append(params);
+        result.append(")");
+
+        return result.toString();
+    }
+
+    public static String encodeParameter(List<Type> parameters) {
```

Comment:
This seems like different with architecture. (decodeReturnParameters function also)
Can you check?